src/pages/page.tsx

```diff
@@ -50,43 +52,55 @@ const PropagatedPage = () => {
   const pageId =
     typeof router.query.id === "string" ? router.query.id : undefined;
   const [title, setTitle] = useState<string>("");
+  const [loading, setLoading] = useState(false);
 
   const fetchOntoPage = useCallback(async () => {
     if (!pageId) {
       return;
     }
-    const page = await apiService.get(`/template/page/${pageId}`);
-    console.log("OntoPage: VIS = ", page.vis);
-    console.log("OntoPage: Data = ", page.data);
-
-    setTitle(page?.title);
-
-    const dataForVisFunction = await Promise.all(
-      page?.data?.map(async (d: any) => {
-        const endpoint = `${API[d.urlCode]}${d.endpoint}`;
-        console.log("PropagatedPage: data endpoint = ", endpoint);
-
-        // const values = (await axios.get(endpoint)).data;
-        const values = (await axios.get(endpoint.replace("data?", "data/?")))
-          .data;
-
-        const { description } = d;
-        return { endpoint, values, description };
-      }),
-    );
-
-    const links = page?.pageIds?.map((d: any) => {
-      console.log(d);
-      return `page?id=${d}`;
-    });
-
-    console.log("PropagatedPage: dataForVisFunction = ", dataForVisFunction);
-
-    visFactory(page?.vis?.function, {
-      chartElement: "charts", // ref.current,
-      data: dataForVisFunction,
-      links,
-    });
+
+    try {
+      setLoading(true);
+
+      const page = await apiService.get(`/template/page/${pageId}`);
+      console.log("OntoPage: VIS = ", page.vis);
+      console.log("OntoPage: Data = ", page.data);
```

Comment:
What is our policy on keeping `console.log`? Just wonder if there is a rule that distinguishes accidentally left statements from the ones we want on the main branch.

Reply:
Here one special requirement is that VIS developers need  console.log for three variables,e.g, data, vis, and links. Other than that there is no policy. Should we delete other console.log?

Reply:
Yeah we could. Leaving a temp `console.log` is a very common mistake, so we can even add https://eslint.org/docs/rules/no-console and explicitly disable permitted `console.log`s:

```ts
// eslint-disable-next-line no-console -- VIS developers need....
console.log("this one is allowed");
```

We can do this in a separate PR.

Reply:
Thanks